dirsrvtests/tests/suites/plugins/attruniq_test.py

```diff
@@ -80,4 +81,68 @@ def test_modrdn_attr_uniqueness(topology_st):
     log.debug(excinfo.value)
 
     log.debug('Move user2 to group1')
-    user2.rename(f'uid={user2.rdn}', group1.dn)
\ No newline at end of file
+    user2.rename(f'uid={user2.rdn}', group1.dn)
+
+    # Cleanup for next test
+    user1.delete()
+    user2.delete()
+
+
+def test_multiple_attr_uniqueness(topology_st):
+    """ Test that attribute uniqueness works properly with multiple attributes
+
+    :id: c49aa5c1-7e65-45fd-b064-55e0b815e9bc
+    :setup: Standalone instance
+    :steps:
+        1. Setup attribute uniqueness plugin to ensure uniqueness of attributes 'mail' and 'mailAlternateAddress'
+        2. Add user with unique 'mail=non-uniq@value.net' and 'mailAlternateAddress=alt-mail@value.net'
+        3. Try adding another user with 'mail=non-uniq@value.net'
+        4. Try adding another user with 'mailAlternateAddress=alt-mail@value.net'
+        5. Try adding another user with 'mail=alt-mail@value.net'
+        6. Try adding another user with 'mailAlternateAddress=non-uniq@value.net'
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Should raise CONSTRAINT_VIOLATION
+        4. Should raise CONSTRAINT_VIOLATION
+        5. Should raise CONSTRAINT_VIOLATION
+        6. Should raise CONSTRAINT_VIOLATION
+    """
+    attruniq = AttributeUniquenessPlugin(topology_st.standalone, dn="cn=multiattruniq,cn=plugins,cn=config")
+
+    try:
+        log.debug(f'Setup PLUGIN_ATTR_UNIQUENESS plugin for {MAIL_ATTR_VALUE} attribute for the group2')
+        attruniq.create(properties={'cn': 'multiattruniq'})
+        attruniq.add_unique_attribute('mail')
```

Comment:
I believe that by default attribute uniqueness plugin configured 'uid'. So those two call add_unique_attribute creates ['uid', 'mail', 'mailAlternateAdress'].
You may call (try) attruniq.remove_unique_attribute('uid')

Reply:
That would apply if a default AttributeUniqueness plugin entry was used, the test however used a new entry. I'll leave this test as is, but I will experiment with modifications to default plugin entry as well.